specs/ScreenCaptureRequested.md

```diff
@@ -0,0 +1,357 @@
+                }
+
+                return S_OK;
+            })
+            .Get(),
+  &m_screenCaptureRequestedToken);
+
+```
+## C++: Registering Screen Requested Handler on CoreWebView2Frame
+``` cpp
+wil::com_ptr<ICoreWebView2> m_webview;
+auto webview4 = m_webview.try_query<ICoreWebView2_4>();
+if (webview4) 
+{
+    EventRegistrationToken m_frameCreatedToken = {};
+    EventRegistrationToken m_screenCaptureRequestedToken = {};
+
+    CHECK_FAILURE(webview4->add_FrameCreated(
+        Callback<ICoreWebView2FrameCreatedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args) 
+              -> HRESULT
+            {
+                wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                CHECK_FAILURE(args->get_Frame(&webviewFrame));
+
+                auto webviewFrame3 = webviewFrame.try_query<ICoreWebView2Frame3>();
+                if (webviewFrame3)
+                {
+                    CHECK_FAILURE(webviewFrame3->add_ScreenCaptureRequested(
+                        Callback<ICoreWebView3FrameScreenCaptureRequestedEventHandler>(
+                            [this](ICoreWebView3Frame* sender,
+                               ICoreWebView2ScreenCaptureRequestedEventArgs* args) -> HRESULT
+                            {
+
+                                    // Get Frame Info
+                                    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo;
+                                    CHECK_FAILURE(args->get_FrameInfo(&frameInfo));
+
+                                    // Frame Source
+                                    wil::unique_cotaskmem_string frameSource;
+                                    CHECK_FAILURE(frameInfo->get_Source(&frameSource));
+
+                                    // If the host app wants to cancel the request for a specific source
+                                    if (frameSource = "https://developer.microsoft.com/en-us/microsoft-edge/webview2/") 
+                                    {
+                                        CHECK_FAILURE(args->put_Cancel(TRUE));
+                                    }
+
+                                    // Let CoreWebView2 handler know the event is already handled
+
+                                    // In the case of an iframe requesting permission to use Screen Capture, the default 
+                                    // behavior is to first raise the ScreenCaptureRequested event off of the 
+                                    // CoreWebView2Frame and invoke it's handlers, and then raise the event off the 
+                                    // CoreWebView2 and invoke it's handlers. However, If we set Handled to true on the
+                                    // CoreWebView2Frame event handler, then we will not raise the
+                                    // ScreenCaptureRequested event off the CoreWebView2.
+
+                                    CHECK_FAILURE(args->put_Handled(true));

```

Comment:
After we set `Handled = true`, what does it mean when Cancel is left as its default value of `false`?

Does it mean "Don't cancel, show default UI"?

Does it mean "Don't cancel, allow silent capture without any prompt"?

Reply:
> Does it mean "Don't cancel, show default UI"?

This is what happens